### PR TITLE
chore: migrate to yarn@2

### PR DESCRIPTION
**Pull request by amilajack**

### CODE REVIEW

**Comment by amilajack** [comment]
Blocked on https://github.com/yarnpkg/berry/issues/2232

**Review by hamirmahal — COMMENTED** [review]
After migrating to yarn 2, are there going to be any special commands people working on this repository will have to run?

I'm assuming, at the very least, running `yarn` at least once after pulling changes from the migration will be necessary.

**Comment by amilajack** [comment]
Nope, that's the nice thing! Same commands. After this is merged, no more need to `rm -rf node_modules && yarn` 🥳

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
             ${{ runner.os }}-yarn-
 
       - name: yarn install
-        run: yarn --frozen-lockfile
+        run: yarn
```
Comment (merceyz) [inline .github/workflows/test.yml:35]: Might want to use the `--immutable` flag https://yarnpkg.com/cli/install#options-immutable

Comment (amilajack) [reply .github/workflows/test.yml]: thanks!

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
             ${{ runner.os }}-yarn-
```
Comment (merceyz) [inline .github/workflows/test.yml:32]: Since you're using zero-installs you could remove the cache step

**Pull request merged**